File: deepspark/spark16/src/main/java/sparklearn/rdd/transformations/Learnmap2.java
/** 
 * Project Name:sparklearn 
 * File Name:Learnmap2.java 
 * Package Name:sparklearn.rdd.transformations 
 * Date:2017年7月4日下午4:40:38 
 * sky.zyl@hotmail.com
*/  
  
package sparklearn.rdd.transformations;

import java.util.Arrays;
import java.util.List;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.api.java.function.VoidFunction;

import com.google.gson.Gson;

/** 
 * ClassName:Learnmap2 <br/> 
 * Function: TODO ADD FUNCTION. <br/> 
 * Reason:   TODO ADD REASON. <br/> 
 * Date:     2017年7月4日 下午4:40:38 <br/> 
 * @author   hadoop 
 * @version   
 * @see       
 */
/**
 * map算子
 * <p>
 * map和foreach算子:
 *   1. 循环map调用元的每一个元素;
 *   2. 执行call函数, 并返回.
 * </p>
 */

public class Learnmap2 {
	
	public static void main(String[] args) {

	    SparkConf conf = new SparkConf().setAppName(Learnmap2.class.getSimpleName())
	            .setMaster("local");

	    JavaSparkContext sc = new JavaSparkContext(conf);

	    List<String> datas = Arrays.asList(
	            "{'id':1,'name':'xl1','pwd':'xl123','sex':2}",
	            "{'id':2,'name':'xl2','pwd':'xl123','sex':1}",
	            "{'id':3,'name':'xl3','pwd':'xl123','sex':2}");

	    JavaRDD<String> datasRDD = sc.parallelize(datas);

	    JavaRDD<User> mapRDD = datasRDD.map(
	            new Function<String, User>() {
	                public User call(String v) throws Exception {
	                    Gson gson = new Gson();
	                    return gson.fromJson(v, User.class);
	                }
	            });

	    mapRDD.foreach(new VoidFunction<User>() {
	        public void call(User user) throws Exception {
	            System.out.println("id: " + user.id
	                    + " name: " + user.name
	                    + " pwd: " + user.pwd
	                    + " sex:" + user.sex);
	        }
	    });
	    
	    sc.close();
	}
	
	class User {
		String id;
		String name;
		String pwd;
		String sex;
		
		public String getId() {
			return id;
		}
		public void setId(String id) {
			this.id = id;
		}
		public String getName() {
			return name;
		}
		public void setName(String name) {
			this.name = name;
		}
		public String getPwd() {
			return pwd;
		}
		public void setPwd(String pwd) {
			this.pwd = pwd;
		}
		public String getSex() {
			return sex;
		}
		public void setSex(String sex) {
			this.sex = sex;
		}

	}
	
}

 

File: deepspark/spark16/src/main/scala/sparklearnscala/rdd/transformations/Learnmap2.scala
package sparklearnscala.rdd.transformations

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import com.google.gson.Gson

object Learnmap2 {
  def main(args: Array[String]) {
    val conf = new SparkConf().setAppName(Learnmap2.getClass.getSimpleName).setMaster("local")
    val sc = new SparkContext(conf)

    val datas: Array[String] = Array(
      "{'id':1,'name':'xl1','pwd':'xl123','sex':2}",
      "{'id':2,'name':'xl2','pwd':'xl123','sex':1}",
      "{'id':3,'name':'xl3','pwd':'xl123','sex':2}")

    sc.parallelize(datas)
      .map(v => {
        new Gson().fromJson(v, classOf[User])
      })
      .foreach(user => {
        println("id: " + user.id
          + " name: " + user.name
          + " pwd: " + user.pwd
          + " sex:" + user.sex)
      })
  }
}
case class User(var id: String,var name: String,var pwd: String,var sex: String) {

}
//class User {
//  var id = ""
//  var name = ""
//  var pwd = ""
//  var sex = ""
//
//  def getId(): String = {
//    return this.id;
//  }
//  def setId(id: String): Unit = {
//    this.id = id;
//  }
//  def getName(): String = {
//    return this.name;
//  }
//  def setName(name: String): Unit = {
//    this.name = name;
//  }
//  def getPwd(): String = {
//    return this.pwd;
//  }
//  def setPwd(pwd: String): Unit = {
//    this.pwd = pwd;
//  }
//  def getSex(): String = {
//    return this.sex;
//  }
//  def setSex(id: String): Unit = {
//    this.sex = sex;
//  }
//
//}